dblookupext/miniblocksHandler.go

```diff
@@ -197,3 +197,88 @@ func (mh *miniblocksHandler) saveMiniblocksMetadata(miniblockHash []byte, minibl
 
 	return mh.miniblocksMetadataStorer.PutInEpoch(miniblockHash, buff, epoch)
 }
+
+func (mh *miniblocksHandler) blockReverted(header data.HeaderHandler, blockBody *block.Body) error {
+	headerHash, errCalculate := core.CalculateHash(mh.marshaller, mh.hasher, header)
+	if errCalculate != nil {
+		return fmt.Errorf("%w for header, header round %d", errCalculate, header.GetRound())
+	}
+
+	for index, mb := range blockBody.MiniBlocks {
+		miniblockHash, err := core.CalculateHash(mh.marshaller, mh.hasher, mb)
+		if err != nil {
+			return fmt.Errorf("%w for miniblock at index %d, header hash %x", err, index, headerHash)
+		}
+
+		err = mh.revertMiniblock(header, headerHash, mb, miniblockHash)
+		if err != nil {
+			return fmt.Errorf("%w for miniblock at index %d, mbHash %x, header hash %x", err, index, miniblockHash, headerHash)
+		}
+	}
+
+	return nil
+}
+
+func (mh *miniblocksHandler) revertMiniblock(header data.HeaderHandler, headerHash []byte, mb *block.MiniBlock, miniblockHash []byte) error {
+	miniblockHeader, err := mh.findMiniblockHandler(header, miniblockHash)
+	if err != nil {
+		return err
+	}
+
+	if miniblockHeader.GetIndexOfFirstTxProcessed() == 0 {
+		// we either revert a complete miniblock or the first partial executed miniblock header
+		err = mh.epochIndex.removeEpochByHash(miniblockHash)
```

Comment:
Nice to spot this cleanup case.

Reply:
right, this is how the miniblock tracker is implemented in the node